engine/src/test/java/org/camunda/bpm/engine/test/api/authorization/history/UserOperationLogAuthorizationTest.java

```diff
@@ -504,6 +513,223 @@ public void testHistoricProcessInstancePermissionsAuthorizationDisabled() {
+
+    // then
+    assertThat(query.list()).isEmpty();
+  }
+
+  public void testCheckReadPermissionOnCompletedHistoricTask() {
+    // given
+    processEngineConfiguration.setEnableHistoricInstancePermissions(true);
+
+    startProcessInstanceByKey(ONE_TASK_PROCESS_KEY);
+    String taskId = selectSingleTask().getId();
+    disableAuthorization();
+    taskService.setAssignee(taskId, userId);
+    taskService.complete(taskId);
+    enableAuthorization();
+
+    createGrantAuthorizationWithoutAuthentication(HISTORIC_TASK, taskId, userId,
+        HistoricTaskPermissions.READ);
+
+    // when
+    UserOperationLogQuery query = historyService.createUserOperationLogQuery();
+
+    // then
+    assertThat(query.list())
+        .extracting("taskId")
+        .containsExactly(taskId, taskId);
+  }
+
+  public void testCheckNonePermissionOnHistoricTaskAndReadHistoryPermissionOnProcessDefinition() {
+    // given
+    processEngineConfiguration.setEnableHistoricInstancePermissions(true);
+
+    startProcessInstanceByKey(ONE_TASK_PROCESS_KEY);
+    String taskId = selectSingleTask().getId();
+    disableAuthorization();
+    taskService.complete(taskId);
+    enableAuthorization();
+
+    createGrantAuthorizationWithoutAuthentication(HISTORIC_TASK, taskId, userId,
+        HistoricTaskPermissions.NONE);
+    createGrantAuthorizationWithoutAuthentication(PROCESS_DEFINITION, ONE_TASK_PROCESS_KEY,
+        userId, ProcessDefinitionPermissions.READ_HISTORY);
+
+    // when
+    UserOperationLogQuery query = historyService.createUserOperationLogQuery();
+
+    // then
+    assertThat(query.list())
+        .extracting("taskId")
+        .containsExactlyInAnyOrder(taskId, null);
+  }
+
+  public void testCheckReadPermissionOnHistoricTaskAndNonePermissionOnProcessDefinition() {
+    // given
+    processEngineConfiguration.setEnableHistoricInstancePermissions(true);
+
+    startProcessInstanceByKey(ONE_TASK_PROCESS_KEY);
+    String taskId = selectSingleTask().getId();
+    disableAuthorization();
+    taskService.complete(taskId);
+    enableAuthorization();
+
+    createGrantAuthorizationWithoutAuthentication(HISTORIC_TASK, taskId, userId,
+        HistoricTaskPermissions.READ);
+    createGrantAuthorizationWithoutAuthentication(PROCESS_DEFINITION, ONE_TASK_PROCESS_KEY, userId,
+        ProcessDefinitionPermissions.NONE);
+
+    // when
+    UserOperationLogQuery query = historyService.createUserOperationLogQuery();
+
+    // then
+    assertThat(query.list())
+        .extracting("taskId")
+        .containsExactly(taskId);
+  }
+
+  public void testCheckNoneOnHistoricTaskAndTaskWorkerCategory() {
+    // given
+    processEngineConfiguration.setEnableHistoricInstancePermissions(true);
+
+    startProcessInstanceByKey(ONE_TASK_PROCESS_KEY);
+
+    String taskId = selectSingleTask().getId();
+    setAssignee(taskId, "demo");
+
+    createGrantAuthorizationWithoutAuthentication(Resources.HISTORIC_TASK,
+        taskId, userId, HistoricTaskPermissions.NONE);
```

Comment:
```suggestion
        taskId, userId, HistoricTaskPermissions.READ);
```

Reply:
I am not sure about this change. The set permissions are not in line with your usual method naming pattern. What should be the goal of the test?